sklearn/cluster/k_means_.py

```diff
@@ -552,17 +553,16 @@ def _labels_inertia_precompute_dense(X, x_squared_norms, centers, distances):
 
     """
     n_samples = X.shape[0]
-    k = centers.shape[0]
-    all_distances = euclidean_distances(centers, X, x_squared_norms,
-                                        squared=True)
-    labels = np.empty(n_samples, dtype=np.int32)
-    labels.fill(-1)
-    mindist = np.empty(n_samples)
-    mindist.fill(np.infty)
-    for center_id in range(k):
-        dist = all_distances[center_id]
-        labels[dist < mindist] = center_id
-        mindist = np.minimum(dist, mindist)
+
+    # Breakup nearest neighbor distance computation into batches to prevent
+    # memory blowup
+    metric_kwargs = dict(squared=True)
+    # Should use check_inputs=False when speedup_kmpp is merged
+    # metric_kwargs = dict(squared=True, check_inputs=False)
+    labels, mindist = pairwise_distances_argmin_min(
+        X=X, Y=centers, metric='euclidean', metric_kwargs=metric_kwargs)
```

Comment:
The default batch size for pairwise_distances_argmin_min is 500. On average it seems that  pairwise_distances_argmin_min is faster than the original implementation. When I add a batch_size to the varied parameters using 500 performs best.